src/docfx/lib/JsonUtility.cs

```diff
@@ -108,14 +111,32 @@ public static (List<Error>, T) ToObject<T>(JToken token)
 
         public static (List<Error>, object) ToObject(JToken token, Type type)
         {
+            ThreadLocal<List<Error>> errors = new ThreadLocal<List<Error>>(() => new List<Error>());
             try
             {
-                return (token.ValidateMismatchingFieldType(type), token.ToObject(type, DefaultDeserializer));
+                var mismatchingErrors = token.ValidateMismatchingFieldType(type);
+                errors.Value.AddRange(mismatchingErrors);
+                DefaultDeserializer.Error += HandleError;
+                _schemaViolationErrors = new List<Error>();
```

Comment:
>_schemaViolationErrors [](start = 16, length = 22)

what about initialize deserializer to pass the errors

Reply:
We should not use the static DefaultDeserializer here, since it might pollute the other methods calls, since we are handling the deserialization errors in a different way.
Since we need to create a new Deserializer for each call, we could pass along the errors

---
In reply to: [209191678](https://github.com/dotnet/docfx/pull/3175#discussion_r209191678) [](ancestors = 209191678)